### PR TITLE
DEPR: Series constructor fastpath keyword

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -374,8 +374,18 @@ def __init__(
         dtype: Dtype | None = None,
         name=None,
         copy: bool | None = None,
-        fastpath: bool = False,
+        fastpath: bool | lib.NoDefault = lib.no_default,
```
Comment (mroeschke) [inline pandas/core/series.py:377]: Do we need/have an alternative for the `fastpath` constructor internally?

Comment (jbrockmendel) [reply pandas/core/series.py:377]: we now use _constructor_from_mgr

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Comment by rhshadrach** [comment]
This patch may have induced a performance regression. If it was a necessary behavior change, this may have been expected and everything is okay.

Please check the links below. If any ASVs are parameterized, the combinations of parameters that a regression has been detected appear as subbullets.

 - [indexing.NumericSeriesIndexing.time_iloc_slice](https://asv-runner.github.io/asv-collection/pandas/#indexing.NumericSeriesIndexing.time_iloc_slice)
   - [dtype=<class 'numpy.float64'>; index_structure='unique_monotonic_inc'](https://asv-runner.github.io/asv-collection/pandas/#indexing.NumericSeriesIndexing.time_iloc_slice?p-dtype=%3Cclass%20%27numpy.float64%27%3E&p-index_structure=%27unique_monotonic_inc%27)

Subsequent benchmarks may have skipped some commits. See the link below to see which commits are between the two benchmark runs where the regression was identified.

https://github.com/pandas-dev/pandas/compare/e22bfb3e1721d602086c0ca0b3af311cbe84cd9b...66a54a37d843cb641bfd4f694946e63a49124b3d

**Comment by lithomas1** [comment]
@jbrockmendel 

Is there an alternative for avoiding name validation now that fastpath is gone? I think this might be what's causing the regressions.

**Comment by jbrockmendel** [comment]
set `result._name = name` instead of passing it to the constructor

**Pull request merged**